standards/SPIFFE-ID.md

```diff
@@ -135,7 +141,47 @@ In order for an SVID to be considered valid, it MUST leverage a document type fo
+
+When arbitrary SVID information is consumed as part of a security decision (such as whether to allow a request or connection), it is critical that the authority issuing the SVID has the same meaning or interpretation of the information as the entity consuming the SVID.
+
+It is a good idea to consider exactly who is operating the authority issuing the SVID. Through out-of-band coordination, trust domain operators may be able to agree on the meaning of a particular piece of information. For example, if an organization operates multiple trust domains for administrative reasons, it may be straightforward to agree on an interpretation of SVID information within that organization (even if the nature of the information is not publicly standardized). The operator should, however, limit processing of that data to only the trust domains with which the meaning has been agreed; agreements of meaning between two trust domain operators should not automatically extend to all trust domain operators.
+
+Example: if trust domains A and B agree to the meaning of the "environment" attribute, that agreement of semantics does not extend to trust domain C which may also happen to include an attribute with the same name of "environment".
+
+#### 4.1.4. Veracity
+SVID consumers residing in the same trust domain as the SVID being consumed can generally assume that assertions in the SVID are true, as the consumer likely considers the authority that issued the SVID to be wholly trusted. It does not necessarily mean however, that assertions of similar form and nature can be assumed true when originating from a different trust domain.
+
+Operators should carefully consider, on a case by case basis, whether or not a given foreign authority should be considered trustworthy in the context of a specific assertion.
+
+For example, imagine that the operator of trust domain A and trust domain B have a shared customer and a business relationship. During normal operation in trust domain A, a specialized process authenticates the customer and the trust domain’s authority creates an SVID that embeds an assertion indicating the customer’s identity. The purpose of this process is to mitigate unauthorized access to customer data in the event that an intermediary service within the trust domain is compromised by proving that the customer was authenticated and did in fact make this request.
+
+Now imagine that a request for the customer’s data is received by the storage service in trust domain A, except the caller presented an SVID from trust domain B. Even though the presented SVID may have the necessary assertion indicating that the shared customer was authenticated and authorized the request, it is a bad idea to blindly assume that trust domain B has indeed authenticated your customer. If trust domain B is compromised, it could claim to have authenticated any user it wishes, thus creating the very circumstance that the measure was designed to mitigate in the first place.
```

Comment:
This example is more charitable to trust domain B than I think is necessary in that it doesn't mention the idea that someone in trust domain B might be lying.

Reply:
IMO "someone in trust domain B is maliciously using the authority" is a subproblem of "trust domain B is compromised". Someone or something has made the signing authority sign something it shouldn't.

I'm not sure how much value is added to the example by exhaustively stating insider vs outsider vs accident vs. malicious etc dimensions to the problem.

However I also don't know that we totally need the full example either. We have a business decision where we trust A to tell us about related customer, and we don't have that same decision with B. Allowing B's processes the same access is overpermissioning, regardless of if such a caller is ever actually present.

Reply:
> However I also don't know that we totally need the full example either. We have a business decision where we trust A to tell us about related customer, and we don't have that same decision with B. Allowing B's processes the same access is overpermissioning, regardless of if such a caller is ever actually present.

I think the real problem here is _who_ you trust to make _which_ assertion. In this case, A should really only trust A to make this kind of user identity assertion. If B tries to make the same assertion, it should be denied. In this way, sure it is a misconfiguration ... however it is an easy one to make/overlook.

This is a very slippery slope in this context. We are saying that maybe we trust the security posture of B enough to trust its SPIFFE IDs and authority... because we are federating. We are further saying that even though we trust B enough to federate, we don't trust it enough to claim that it has authenticated one of our customers and then access their data. And it is this point I was trying to highlight with the example. The veracity decision is not applied equally to all assertions, it must be made on a case-by-case basis.

I think it's an important distinction to make, but if I'm muddying the waters with this example I'd love some suggestions on how to make this clearer.

Reply:
I think the overall example is fine; it's just the last sentence reads to me to imply that trust domain B being compromised is the main item in the threat model.  The improvement I would suggest is, instead of giving an example of why you might not want to trust domain B to make this assertion, we instead start from zero trust and ask the reader to consider why they should trust that assertion.  Is there an operational need to trust it?  If so, do they understand how trust domain B is making that assertion?  Does the customer who's data it is agree?

Reply:
I added a small blurb here that B may also have a malicious internal actor.